Remove duplicates from common-definitions

phackstock opened the pull request:
Closes #1.

danielhuppmann commented on definitions/variable/variable.yaml:
```diff
@@ -448,7 +448,7 @@
 - Temperature|Global Mean:
     definition: change in global mean temperature relative to pre-industrial
-    unit: '°C'
+    unit: "\xB0C"
```
please keep readable strings

phackstock replied:
Done

danielhuppmann reviewed (changes requested):
The variables "Final Energy|Carbon Removal|Electricity|Direct Air Capture" etc. were renamed for consistency (to have the fuel as last entry), see [here](https://github.com/IAMconsortium/common-definitions/blob/26be13dc32b7179f1b55000fd28a62006ad1be31/definitions/variable/energy/final-energy.yaml#L117) - please remove the almost-duplicate variables here as well.

phackstock commented:
> The variables "Final Energy|Carbon Removal|Electricity|Direct Air Capture" etc. were renamed for consistency (to have the fuel as last entry), see [here](https://github.com/IAMconsortium/common-definitions/blob/26be13dc32b7179f1b55000fd28a62006ad1be31/definitions/variable/energy/final-energy.yaml#L117) - please remove the almost-duplicate variables here as well.

Removed them and fixed the degree formatting.

danielhuppmann reviewed (approved):
Looks good, thanks! I just realized that common-definitions does not have "Population|Rural" (but urban population is included), so we could use this to discuss an integration-test-workflow to flag any duplicates as we develop common-definitions further...

phackstock commented:
> Looks good, thanks! I just realized that common-definitions does not have "Population|Rural" (but urban population is included), so we could use this to discuss an integration-test-workflow to flag any duplicates as we develop common-definitions further...

I had the same thought, it's probably worth while looking at the variables that are left in legacy-definitions to check if there's any that we should move over. Such as the "Population|Rural" that you pointed out.

Fully agree on the integration test. I already quickly looked into this. My idea was that the tests for legacy-definitions would run on every commit on main to common-definitions. Unfortunately, the functionality to trigger a workflow from one repository to a different one does not exist natively in GitHub. There are options to manually send a trigger in a GitHub action but it'll probably require a bit more fiddling.

Actually, now that I think about it we should probably remove the `nomenclature.yaml` file from this repo. Reason being that, as it stands, it imports common-definitions. This was very convenient up until now since it allowed us to check for the compatibility of the two definitions.
However, for the future I would imagine that a lot of projects would use a `nomenclature.yaml` like so:

```yaml
repositories:
  common-definitions:
    url: https://github.com/IAMconsortium/common-definitions.git/
  legacy-definitions:
    url: https://github.com/IAMconsortium/legacy-definitions.git/
definitions:
  variable:
    repository: 
      - common-definitions
      - legacy-definitions
```

This would, as it currently stands fail since legacy-definitions currently **contains** common-definitions so the above setup would double common-definitions leading to artificial duplicate code entries.
